src/mbedtls.c

```diff
@@ -119,14 +119,28 @@ _libssh2_mbedtls_cipher_init(_libssh2_cipher_ctx *ctx,
     if(!ctx)
         return -1;
 
-    op = encrypt == 0 ? MBEDTLS_ENCRYPT : MBEDTLS_DECRYPT;
+    op = encrypt ? MBEDTLS_ENCRYPT : MBEDTLS_DECRYPT;
```

Comment:
Makes me wonder how could libssh2 + mbedTLS work correctly before if this condition
was reversed all along?